src/targets/gpu/jit/reduce.cpp

```diff
@@ -145,18 +157,29 @@ struct simple_reduce_compiler : compiler<simple_reduce_compiler>
         auto faxis             = find_fast_axis({options.virtual_inputs.front()});
         vectorize vec{};
         auto nelements = options.virtual_inputs.back().elements();
-        auto algo      = v.get("algo", get_reduce_algo(options.virtual_inputs));
-        if(algo == "block")
+        auto algo      = v.get("algo", get_reduce_algo(ctx, options.virtual_inputs));
```

Comment:
I think it would be better to get the wavefront size before calling into all of these functions. Then pass in the wavefront size everywhere else instead of `ctx`.

Reply:
We could depend on other hardware attributes in the future, so I dont really want to expose such details in the function interface.